### ISSUE TITLE
Seurat processed dataset as input into MultiK

### ISSUE THREAD

**Issue by strawberry789** [siyao-liu/MultiK#2]
How can I use a processed single cell dataset from Seurat (before clustering) as an input into MultiK?

**Comment by siyao-liu** [comment]
@strawberry789  Hi! Because MultiK re-selects genes in each subsampling run, you will need to provide a normalized single cell dataset from Seurat as an input in order to run MultiK. If your processed single cell dataset from Seurat is already normalized, you should be able to use it as an input into MultiK. Thanks!

**Comment by sthyme** [comment]
I saw that you also do a NormalizeData call within the getClusters.R. Does this mean that the data should be normalized 2x? Before the MultiK call and then again in the getClusters call? Also, is it possible to use with SCTransform normalized data?

**Comment by siyao-liu** [comment]
@sthyme No, the data shouldn't be normalized twice. The input data for the getClusters.R function should be an unnormalized count matrix.
I think you can use SCTransform normalized data. The MultiK function only requires a normalized data matrix. Thanks!

**Comment by njardiss** [comment]
@siyao-liu Thank you for this package Siyao Liu.

When I call MultiK() or getClusters() functions with a seurat object does the methods use the current default assay like RNA, SCT or integrated or does this have to be specified in some way?

I have a dataset from several donors and these are integrated together to reduce batch effect. I assume that MultiK has to be run on the integrated assay to avoid the batch/donor effect, is this right? 


Is getClusters() compatible with the Seurat vignette for SCTransform integrated dataset or is there another way to use MultiK to find better clusters for these integrated datasets?
https://satijalab.org/seurat/articles/integration_introduction.html#performing-integration-on-datasets-normalized-with-sctransform-1

Thanks!

**Comment by siyao-liu** [comment]
@njardiss Hi! Sorry about the delay in responding to you. Currently, only the default RNA assay is used in MultiK() and getClusters().

Unfortunately, current version of MultiK does not support integrated datasets. We are working on a new function that would run on integrated datasets. Please stay tuned! Thanks!

**Comment by Linpu2023** [comment]
Hi @njardiss, I encountered error when running getClusters() on SCTransformed object (default assay is SCT):

`clusters <- getClusters(Ctrl_OFE, 3)`
`Error in names(x) <- value : 
  'names' attribute [1] must be the same length as the vector [0]`

I just saw your discussion and the code works if I reset the default assay to RNA.

I have another question, how do I know if the MultiK() is applied on the "RNA" assay or "SCT"?
